### PR TITLE
Adding HappyBase Connection.create_table().

### PR DESCRIPTION
**Pull request by dhermes**
@tseaver sorry #1501 got closed. I did a `git push -f` after rebase-ing and it just pushed the `HEAD` of `master` (which GitHub interpreted as a delete).

### CODE REVIEW

File: gcloud/bigtable/happybase/connection.py [hunk 1]
```diff
@@ -264,6 +269,77 @@ def tables(self):
...
+
+        :type name: str
+        :param name: The name of the table to be created.
+
+        :type families: dict
+        :param families: Dictionary with column family names as keys and column
+                         family options as the values. The options can be among
+
+                         * :class:`dict`
+                         * :class:`.GarbageCollectionRule`
```
Comment (dhermes) [inline gcloud/bigtable/happybase/connection.py]: @tseaver For posterity, the concern from #1501 was right here. From you

> That feels wrong, somehow: how does one spell options for a CF which has both GC rules and other settngs?

and from me

----

We are [trying to emulate][1] dictionary only (in HappyBase):

```python
families = {
    'cf1': dict(max_versions=10),
    'cf2': dict(max_versions=1, block_cache_enabled=False),
    'cf3': dict(),  # use defaults
}
connection.create_table('mytable', families)
```

but Cloud Bigtable has more / different features than HappyBase for garbage collection rules, so I wanted to expand the allowed values.

[1]: https://github.com/wbolster/happybase/blob/9cbd718c10a3089f234f1eac1236b631e1f8e7cd/happybase/connection.py#L248-L265

**Comment by tseaver** [comment]
Do all the possible options (dict keys) control only garbage collection?

**Comment by dhermes** [comment]
The [only fields][1] for a table (other than a `name`) are the GC rules. So the keys of `families` the dictionary (the one that gets passed to `create_table`) are the names of the column families (as strings) and then the values specify the configuration for each table.

You're right in wondering though. I'd imagine if at some point the Thrift/HBase table settings

```
max_versions
compression
in_memory
bloom_filter_type
bloom_filter_vector_size
bloom_filter_nb_hashes
block_cache_enabled
time_to_live
```

then we'd have a re-design on our hands.

[1]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/aa4064ec48b31c09fe11c4be4329318a001452e9/bigtable-protos/src/main/proto/google/bigtable/admin/table/v1/bigtable_table_data.proto#L53

**Comment by dhermes** [comment]
How big of a n00b am I! I closed this by accident (from the command line) the exact same way I closed #1501.

**Comment by dhermes** [comment]
@tseaver Does my [explanation][1] make sense?

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/pull/1502#issuecomment-186463337

File: gcloud/bigtable/happybase/connection.py [hunk 2]
```diff
@@ -336,3 +411,51 @@ def compact_table(self, name, major=False):
...
+    :rtype: :class:`.GarbageCollectionRule`
+    :returns: A garbage collection rule parsed from the input.
+    :raises: :class:`ValueError <exceptions.ValueError>` if ``option`` is a
+             dictionary but keys other than ``max_versions`` and
+             ``time_to_live`` are used.
+    """
+    result = option
+    if isinstance(result, dict):
+        if not set(result.keys()) <= set(['max_versions', 'time_to_live']):
+            raise ValueError('Cloud Bigtable only supports max_versions and '
```
Comment (jonparrott) [inline gcloud/bigtable/happybase/connection.py]: Hmm, it seems in other places we warn in the name of compatibility with HappyBase, but here we're actually throwing an error. Is that intended?

File: gcloud/bigtable/happybase/connection.py [hunk 3]
```diff
@@ -265,6 +269,77 @@ def tables(self):
...
+        """Create a table.
+
+        .. warning::
+
+            The only column family options from HappyBase that are able to be
+            used with Cloud Bigtable are ``max_versions`` and ``time_to_live``.
+
+        .. note::
+
+            This method is **not** atomic. The Cloud Bigtable API separates
```
Comment (jonparrott) [inline gcloud/bigtable/happybase/connection.py]: This is concerning. Can we provide any recommendations for recourse for the user? Or is it expected that they'll know how to deal with the broken state?

File: gcloud/bigtable/happybase/connection.py (hunk 2 above)
Comment (dhermes) [inline gcloud/bigtable/happybase/connection.py]: Good call.

Comment (dhermes) [inline gcloud/bigtable/happybase/connection.py]: I originally raised everywhere but changed philosophy after chatting with @jgeewax

File: gcloud/bigtable/happybase/connection.py (hunk 3 above)
Comment (dhermes) [inline gcloud/bigtable/happybase/connection.py]: @jgeewax Who should we reach out to about this question? HBase/Thrift allow the columns to created when the table is, but [`CreateTableRequest`][1] does not.

To deal with it here we could also try to clean up, e.g. if there is any failure during creation, we try to delete all the things that were created. But IMO that is opening Pandora's box.

[1]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/7c77c170c8818a0440e0d2f4ecedcb8329c1f8d9/bigtable-protos/src/main/proto/google/bigtable/admin/table/v1/bigtable_table_service_messages.proto#L26

**Comment by dhermes** [comment]
@jonparrott I changed the exception to a warning in `_parse_family_option`. Still waiting on the atomicity / rollback question.

**Comment by jonparrott** [comment]
LGTM, barring the question about atomicity. 

(don't forget to squash)

**Comment by dhermes** [comment]
OK. Will squash and merge and file an issue to follow up.

**Comment by jonparrott** [comment]
SGTM.

**Pull request merged**